packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -526,74 +610,236 @@ export const Tabs = (props: TabsProps) => {
+});
+TabIcon.displayName = 'Tabs.TabIcon';
+
+// ----------------------------------------------------------------------------
+// Tabs.TabLabel component
+
+const TabLabel = polymorphic.span('iui-tab-label');
+TabLabel.displayName = 'Tabs.TabLabel';
+
+// ----------------------------------------------------------------------------
+// Tabs.TabDescription component
+
+const TabDescription = React.forwardRef((props, ref) => {
+  const { className, children, ...rest } = props;
+  const { type } = useSafeContext(TabsContext);
+  const { hasSublabel, setHasSublabel } = useSafeContext(TabListContext);
+
+  useIsomorphicLayoutEffect(() => {
+    type !== 'pill' && !hasSublabel && setHasSublabel && setHasSublabel(true);
+  }, []);
+
+  if (type !== 'pill') {
+    return (
       <Box
-        as='ul'
-        className={cx(
-          'iui-tabs',
-          `iui-${type}`,
-          {
-            'iui-green': color === 'green',
-            'iui-animated': type !== 'default' && isClient,
-            'iui-not-animated': type !== 'default' && !isClient,
-            'iui-large': hasSublabel,
-          },
-          tabsClassName,
-        )}
-        data-iui-overflow={overflowOptions?.useOverflow}
-        data-iui-scroll-placement={scrollingPlacement}
-        role='tablist'
-        ref={refs}
-        onKeyDown={onKeyDown}
+        as='span'
+        className={cx('iui-tab-description', className)}
+        ref={ref}
         {...rest}
       >
-        {labels.map((label, index) => createTab(label, index))}
+        {children}
       </Box>
+    );
+  } else {
+    return <></>;
+  }
+}) as PolymorphicForwardRefComponent<'span'>;
+TabDescription.displayName = 'Tabs.TabDescription';
 
-      {actions && (
-        <Box className='iui-tabs-actions-wrapper'>
-          <Box className='iui-tabs-actions'>{actions}</Box>
-        </Box>
-      )}
+// ----------------------------------------------------------------------------
+// Tabs.Actions component
+
+const TabsActions = React.forwardRef((props, ref) => {
+  const { className, children, ...rest } = props;
+
+  const { type } = useSafeContext(TabsContext);
+
+  if (type !== 'pill') {
+    return (
+      <Box className={cx('iui-tabs-actions', className)} ref={ref} {...rest}>
+        {children}
+      </Box>
+    );
+  } else {
+    return <></>;
+  }
+}) as PolymorphicForwardRefComponent<'div'>;
+TabsActions.displayName = 'Tabs.Actions';
+
+// ----------------------------------------------------------------------------
+// Tabs.Panel component
+
+type TabsPanelOwnProps = {
+  /**
+   * Value used to associate the panel with a given tab.
+   */
+  value: string;
+};
+
+const TabsPanel = React.forwardRef((props, ref) => {
+  const { value, className, children, ...rest } = props;
+
+  const { activeValue } = useSafeContext(TabsContext);
+
+  if (value === activeValue) {
+    return (
+      <Box
+        className={cx('iui-tabs-content', className)}
+        role='tabpanel'
+        id={value}
```

Comment:
we could prefix this value with a unique id (unique and shared throughout the whole Tabs) in case the user uses two sets of tabs with overlapping ids.

also i believe we will need to account for the user passing `id` directly